### DISCUSSION TITLE
JS Hook accessing record properties is undefined.

### DISCUSSION THREAD

**Discussion by jcmiron** [pocketbase/pocketbase#4912] [Q&A] [answered]
I can't for the life of me figure out why accessing a record property is undefined in my custom route. (data scrubbed)

In my custom route, I added a bunch of $app.logger() to try to troubleshoot this. 

```
$app.logger().info('/custom/' + slug);  // outputs /custom/hello

const record = $app.dao().findFirstRecordByFilter('collection', "(Slug ~ 'hello')");

$app.logger().info('Retrieved record: ' + JSON.stringify(record)); // { "ClassName":"HelloClass","Slug":"hello" }    
$app.logger().info('Accessing ClassName directly: ' + record.ClassName); // Accessing ClassName directly: undefined
$app.logger().info('Accessing ClassName with bracket notation: ' + record['ClassName']); //Accessing ClassName with bracket notation: undefined
```

I'm originally trying to use the record.ClassName to power 2 more queries, but in the logs of pocketbase it shows that the parameter is "undefined"

I tried wrapping the calls in a transaction, still no dice. I've tried every combo that I can find on the docs of PocketBase and I'm running out of ideas.

**Comment by ganigeorgiev** [comment, marked as answer]
The `Record` model is a translated version of the Go [`models.Record`](https://pkg.go.dev/github.com/pocketbase/pocketbase/models#Record) struct, aka. it is not a POJO and to access the schema fields you need to use the designated getter/setters - https://pocketbase.io/docs/js-records/#getset-record-fields:

```js
record.get("ClassName")
```

**Reply by jcmiron** [reply]
That did it! I'm so mad at myself, I saw that page and those items many times today and because of the examples below it I just assumed it was for something else....

Thanks!
